### PR TITLE
Feat queue and worker: add missing event typings

### PR DESCRIPTION
**Pull request by roggervalf**
Same as the title

### CODE REVIEW

File: src/classes/queue.ts [hunk 1]
```diff
@@ -209,7 +214,7 @@ export class Queue<
       | 'paused'
       | 'delayed'
       | 'failed' = 'completed',
-  ) {
+  ): Promise<string[]> {
```
Comment (manast) [inline src/classes/queue.ts:217]: This change is breaking a couple of tests.

Comment (roggervalf) [reply src/classes/queue.ts:217]: Thank you, I updated it, now should be fine for another :eyes:

File: src/test/test_worker.ts [hunk 2]
```diff
@@ -240,11 +241,11 @@ describe('workers', function() {
       expect(job.data.foo).to.be.eql('bar');
 
       return new Promise(resolve => {
-        worker.on('failed', async jobId => {
+        worker.on('failed', async (job, error) => {
```
Comment (roggervalf) [inline src/test/test_worker.ts:244]: I changed these test cases because when digging into those, I found that job were instances and not id strings

Comment (manast) [reply src/test/test_worker.ts:244]: True, local events on workers returns instances, this is the same as in Bull v3.

**Review by manast — APPROVED** [review]

**Pull request merged**